### PR TITLE
feat(aws): Add support for AWS Backup resource

### PR DESCRIPTION
**Pull request by ekifel**
Issue #344 

Details: 

- I didn't find cost component for `AWS Storage Gateway Volume Backup`, searched by services, prices - nothing. 
- Didn't add part with cross-region data transfer as we already have resource `aws_data_transfer` which cover all of that functionality
- I compared prices all of non-usage cost components (`Monthly cost depends on...`) from Golden file with [AWS Pricing page](https://aws.amazon.com/backup/pricing/), they are equals

### CODE REVIEW

**Comment by ekifel** [comment]
Hello @tim775, can you take a look on this as you already investigated this resource?

**Comment by tim775** [comment]
@alikhajeh1 LGTM

File: internal/providers/terraform/aws/backup_vault.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		{ref: "monthly_efs_cold_restore_gb", name: "EFS restore (cold)", unit: "GB", usageType: "PartialRestore-Cold-EFS", service: "AWSBackup", family: "AWS Backup Storage"},
+		{ref: "monthly_efs_item_restore_requests", name: "EFS restore (item-level)", unit: "requests", usageType: "PartialRestore-Jobs-EFS", service: "AWSBackup", family: "AWS Backup Storage"},
+		{ref: "monthly_ebs_snapshot_gb", name: "EBS snapshot", unit: "GB", usageType: "EBS:SnapshotUsage$", service: "AmazonEC2", family: "Storage Snapshot"},
+		{ref: "monthly_rds_snapshot_gb", name: "RDS snapshot", unit: "GB", usageType: "RDS:ChargedBackupUsage", service: "AmazonRDS", family: "Storage Snapshot"},
+		{ref: "monthly_dynamodb_backup_gb", name: "DynamoDB backup", unit: "GB", usageType: "TimedBackupStorage-ByteHrs", service: "AmazonDynamoDB", family: "Amazon DynamoDB On-Demand Backup Storage"},
+		{ref: "monthly_dynamodb_restore_gb", name: "DynamoDB restore", unit: "GB", usageType: "RestoreDataSize-Bytes", service: "AmazonDynamoDB", family: "Amazon DynamoDB Restore Data Size"},
+		//	{ref: "monthly_storage_gateway_backup_gb", name: "Storage gateway backup", unit: "GB", usageType: "", service: "", family: ""},
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/backup_vault.go:44]: why is this one commented out?

**Review by alikhajeh1 — COMMENTED** [review]
Thanks @ekifel and @tim775 - one tiny question then we can merge this

File: internal/providers/terraform/aws/backup_vault.go (hunk 1 above)
Comment (tim775) [reply internal/providers/terraform/aws/backup_vault.go:44]: Yeah @ekifel initial comment on the PR states that he couldn't find them.  Adding an explanation to the comment in the code is a good idea.

**Comment by alikhajeh1** [comment]
@ekifel can you plz add the other resources in the service to the free list? 
Screen Shot 2021-06-22 at 9 06 16 AM

**Comment by ekifel** [comment]
@alikhajeh1, sure! Added

**Pull request merged**